src/components/Trades/EditTradeModal.js

```diff
@@ -0,0 +1,198 @@
+      },
+      {
+        header: "Currency",
+        type: "text",
+        value: currencyField,
+        setter: setCurrencyField,
+      },
+      {
+        header: "Platform",
+        type: "text",
+        value: platformField,
+        setter: setPlatformField,
+      },
+    ];
+    const editFields = fieldList.map((item) => (
+      <Form.Group controlId="date" bssize="large" key={item.header}>
+        <Form.Label>{item.header}</Form.Label>
+        {item.type ? (
+          <Form.Control
+            defaultValue={item.value}
+            type={item.type}
+            onChange={(e) => handleChange(e, item.setter)}
+          />
+        ) : (
+          <select
+            className="action-select"
+            value={item.value}
+            onChange={(e) => handleChange(e, item.setter)}
+          >
+            <option value="Buy">Buy</option>
+            <option value="Sell">Sell</option>
+          </select>
+        )}
+      </Form.Group>
+    ));
+    setEditFields(editFields);
+  }, [
+    nameField,
+    codeField,
+    buySellField,
+    dateField,
+    priceField,
+    currencyField,
+    platformField,
+  ]);
+
+  const handleChange = (e, setter) => {
+    setter(e.target.value);
+  };
+
+  const setFields = () => {
+    const date = new Date(props.trade.val.date);
+    setNameField(props.trade.val.stockName);
+    setCodeField(props.trade.val.stockCode);
+    setBuySellField(props.trade.val.action);
+    setDateField(
+      `${date.toLocaleDateString("en-GB", {
+        year: "numeric",
+      })}-${date.toLocaleDateString("en-GB", {
+        month: "numeric",
+      })}-${date.toLocaleDateString("en-GB", { day: "numeric" })}`
+    );
+    setPriceField(props.trade.val.price);
+    setCurrencyField(props.trade.val.currency);
+    setPlatformField(props.trade.val.platform);
+  };
+
+  const resetFields = () => {
+    setNameField("");
+    setCodeField("");
+    setBuySellField("");
+    setDateField("");
+    setPriceField("");
+    setCurrencyField("");
+    setPlatformField("");
+  };
+
+  const handleClose = () => {
+    resetFields();
+    props.close();
+  };
+
+  const editTrade = () => {
+    const tradeDate = new Date(dateField) - 0;
+    const updatedTrade = {
+      stockCode: codeField,
+      stockName: nameField,
+      action: buySellField,
+      date: tradeDate,
+      price: priceField,
+      currency: currencyField,
+      platform: platformField,
+    };
+    const tradeRef = ref(database, `${user}/${TRADES_KEY}/${props.trade.key}`);
+    update(tradeRef, updatedTrade).then(() => {
+      let newTradesArr = props.tradesArr;
+      for (let i = 0; i < newTradesArr.length; i++) {
+        if (newTradesArr[i].key === props.trade.key) {
+          newTradesArr.splice(i, 1);
+          newTradesArr.push({ key: props.trade.key, val: updatedTrade });
+          props.setTradesArr([...newTradesArr]);
+          break;
+        }
+      }
+      props.close();
+    });
+  };
+  return (
```

Comment:
this component seems very similar to `AddTradeLineItem`, possibly able to refactor and create reusable code?